feat(github): set package registry credentials automatically (#25214)
renovatebot/renovate: lib/modules/platform/github/index.md

It needs to have the user id _and_ the username followed by the `users.noreply.`-domain of either github.com or the GitHub Enterprise Server.
A way to get the user id of a GitHub app is to [query the user API](https://docs.github.com/en/rest/reference/users#get-a-user) at `api.github.com/users/self-hosted-renovate[bot]` (github.com) or `github.enterprise.com/api/v3/users/self-hosted-renovate[bot]` (GitHub Enterprise Server).

## Package Registry Credentials

> When Renovate is running against repos on `github.com`, it will automatically provision `hostRules` for these GitHub Packages registries using the platform token:

Reviewer: ```suggestion
When Renovate runs against repositories on `github.com`, it automatically provisions `hostRules` for these GitHub Packages registries using the platform token:
```
